src/Driver/Driver.hs

```diff
@@ -218,8 +219,10 @@ inferDecl mn (Core.CmdDecl decl) = do
 --
 inferDecl mn (Core.DataDecl decl) = do
   -- Insert into environment
-  let f env = env { declEnv = (RST.data_loc decl,decl) : declEnv env, kindEnv = insertKinds decl (kindEnv env)}
-
+  let loc = RST.data_loc decl
+  env <- gets drvEnv
+  decl' <- liftEitherErrLoc loc (resolveDataDecl decl env)
+  let f env = env { declEnv = (loc, decl') : declEnv env, kindEnv = insertKinds decl (kindEnv env)}
   modifyEnvironment mn f
   pure (TST.DataDecl decl)
```

Comment:
Warum `pure (TST.DataDecl decl)` und nicht `(pure TST.DataDecl decl')`?

Reply:
Sprich: ist es Absicht, dass in `TST.Program` noch `RST.DataDecl` verwendet wird?

Reply:
Eigentlich sollte das schon TST sein, das ist jetzt geändert